mod_ci/controllers.py

```diff
@@ -1381,6 +1372,16 @@ def progress_type_request(log, test, test_id, request) -> bool:
     except GithubException as a:
         log.error(f'Got an exception while posting to GitHub! Message: {a.data}')
 
+    if status in [TestStatus.completed, TestStatus.canceled]:
```

Comment:
Can you help me understand how changing the position of the code in the file is fixing the issue? 🤔

Reply:
Actually, the only change I found I did in this function was to add this, and I feel the wait_for_operation function might be blocking, locally the whole function works fine; the only function I couldn't test was of deleting instances.

Anyways we don't have enough logs as well to debug this past through, so adding logging and moving this function at the end might help us to debug this